### DISCUSSION TITLE
What is the difference between shortest_simple_paths and all_shortest_paths?

### DISCUSSION THREAD

**Discussion by TOMOFUMI-KONDO** [networkx/networkx#6381] [Q&A] [answered]
I searched the method to find k the shortest paths in the networkx graph, and found functions named '[shortest_simple_paths](https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.simple_paths.shortest_simple_paths.html)' and '[all_shortest_paths](https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.shortest_paths.generic.all_shortest_paths.html#all-shortest-paths)'. It seems that Both of them are used to find all the shortest paths between specified source and target nodes. What is the difference of those methods? How I decide which to use.

**Comment by dschult** [comment, marked as answer]
The function `shortest_simple_paths` yields all the simple paths in order of length.  So most of the resulting paths are not "shortest paths".  To explain the name, think of "I want the 6 shortest simple paths".

The function `all_shortest_paths` yields all of the possible shortest paths. All of these paths will have the same length. The use for this function is when there are multiple ways to go that are the same distance. Instead of just giving one example of a shortest path, this function provides them all.

Perhaps the docs should include a comparison of these functions...  Anyone know if a paragraph be included in a Sphinx "See Also" section?

**Reply by TOMOFUMI-KONDO** [reply]
Thank you. I understood the difference of two functions.
I agree that docs should include a comparison of them.

**Comment by rossbar** [comment]
> Perhaps the docs should include a comparison of these functions... Anyone know if a paragraph be included in a Sphinx "See Also" section?

A text description can be added to entries in the `See Also` section, though I think the intent is to support short descriptions rather than paragraphs. That said, more documentation on how the shortest path functions related to one another would certainly be useful. Another potential place to beef up documentation would be the `shortest_path` module/package docstring. IMO it'd also be a great candidate for an `nx-guide` to provide a more in-depth, executable example to highlight what each of the functions does.
